sklearn/metrics/_scorer.py

```diff
@@ -296,6 +302,13 @@ def _score(self, method_caller, clf, X, y, sample_weight=None):
             y_pred = method_caller(clf, "predict", X)
         else:
             try:
+                if (
+                    y_type == "binary"
```

Comment:
So here, we could have a `ScorerProperty` defining that the score is symmetric and require `pos_label` instead of hard coding `roc_auc_score`